### PR TITLE
Switch to 526 Save-in-progress endpoint

### PR DESCRIPTION
**Pull request by Mottie**
## Description

Form 526 has a new save-in-progress endpoint - [`/v0/disability_compensation_in_progress_forms/`](https://department-of-veterans-affairs.github.io/va-digital-services-platform-docs/api-reference/#/in_progress_forms/getInProgressForm1) which behaves almost exactly like the original `/v0/in_progress_forms/` endpoint except that it dynamically updates the `ratedDisabilities` values to provide the most up-to-date values. Additionally, when the `ratedDisabilities` values change, a `ratedDisabilitiesUpdated: true` flag is added to the metadata and the user is returned to the rated disability page of the form (in the metadata `returnUrl`). 

Related:
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/13128#issuecomment-763838119
- BE: https://github.com/department-of-veterans-affairs/vets-api/pull/5717
- Swagger: https://department-of-veterans-affairs.github.io/va-digital-services-platform-docs/api-reference/#/in_progress_forms/getInProgressForm1

## Testing done

Updated unit tests
Updated 526 Cypress tests

## Screenshots

N/A

## Acceptance criteria
- [x] Switch 526 save-in-progress to new API endpoint
- [x] All tests passing

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms/save-in-progress/actions.js [hunk 1]
```diff
@@ -251,12 +252,14 @@ export function fetchInProgressForm(
   //  redux store, but form.migrations doesn’t exist (nor should it, really)
   return (dispatch, getState) => {
     const trackingPrefix = getState().form.trackingPrefix;
+    const apiUrl =
+      VA_FORM_IDS_IN_PROGRESS_FORMS_API[formId] || '/v0/in_progress_forms/';
```
Comment (cvalarida) [inline src/platform/forms/save-in-progress/actions.js:256]: How do you feel about standardizing this lookup in a function which does just this? Then we can do something like

```js
const apiUrl = inProgressApi(formId);
```

It'll then consolidate the `/v0/in_progress_forms/` magic string to one place.

File: src/platform/forms/tests/save-in-progress/actions.unit.spec.js [hunk 2]
```diff
@@ -166,6 +166,21 @@ describe('Schemaform save / load actions:', () => {
           done(err);
         });
     });
+    it('calls the disability compensation api to save the form', done => {
```
Comment (cvalarida) [inline src/platform/forms/tests/save-in-progress/actions.unit.spec.js:169]: I recommend making this test name a little more generic to describe what it's really doing. For now, it's just the disability compensation API, but the _functionality_ is more like `calls the overridden API endpoint for saving forms with an override` (or something).

File: src/platform/forms/tests/save-in-progress/actions.unit.spec.js [hunk 3]
```diff
@@ -320,6 +335,27 @@ describe('Schemaform save / load actions:', () => {
...
       });
     });
+    it('dispatches a success if the form is loaded', () => {
+      const thunk = fetchInProgressForm(VA_FORM_IDS.FORM_21_526EZ, {});
+      const dispatch = sinon.spy();
+      global.fetch.returns(
+        Promise.resolve({
+          ok: true,
+          json: () => ({
+            formData: { field: 'foo' }, // eslint-disable-line camelcase
```
Comment (cvalarida) [inline src/platform/forms/tests/save-in-progress/actions.unit.spec.js:345]: Is this `eslint-disable-line` needed?

File: src/platform/forms/tests/save-in-progress/actions.unit.spec.js [hunk 4]
```diff
@@ -320,6 +335,27 @@ describe('Schemaform save / load actions:', () => {
         );
       });
     });
+    it('dispatches a success if the form is loaded', () => {
```
Comment (cvalarida) [inline src/platform/forms/tests/save-in-progress/actions.unit.spec.js:338]: Like above, this is specifically testing that the _override_ works, right? Can we make the test name reflect this?

**Review by cvalarida — APPROVED** [review]
This looks good from the forms library side of things. I recommend a little cleanup, but it's not enough to stop me from approving. :+1:

File: src/platform/forms/save-in-progress/actions.js (hunk 1 above)
Comment (Mottie) [reply src/platform/forms/save-in-progress/actions.js:256]: Sounds like a good idea!

File: src/platform/forms/tests/save-in-progress/actions.unit.spec.js (hunk 3 above)
Comment (Mottie) [reply src/platform/forms/tests/save-in-progress/actions.unit.spec.js:345]: I don't think so... did I add that? 😱

Comment (Mottie) [reply src/platform/forms/tests/save-in-progress/actions.unit.spec.js:345]: LOL, I copy/pasted it from the test above 😛

**Review by micahchiang — APPROVED** [review]

**Pull request merged**